soc/xtensa/intel_adsp/common/mem_window.c

```diff
@@ -0,0 +1,56 @@
+/* Copyright(c) 2022 Intel Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/arch/xtensa/cache.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+
+#include <adsp_memory.h>
+#include <adsp_shim.h>
+#include <mem_window.h>
+#include <soc.h>
+
+/* host windows */
+#define DMWBA(win_base) (win_base + 0x0)
+#define DMWLO(win_base) (win_base + 0x4)
+
+#define DT_DRV_COMPAT intel_adsp_mem_window
+
+#define WIN_SIZE(N)   (CONFIG_MEMORY_WIN_##N##_SIZE)
+#define WIN0_OFFSET DT_INST_PROP(0, offset)
+#define WIN1_OFFSET WIN0_OFFSET + WIN_SIZE(0)
+#define WIN2_OFFSET WIN1_OFFSET + WIN_SIZE(1)
+#define WIN3_OFFSET WIN2_OFFSET + WIN_SIZE(2)
```

Comment:
This is probably misuse of the INST APIs. Did you read https://docs.zephyrproject.org/latest/build/dts/api/api.html#c.DT_INST? Instance 0 is not necessarily the same thing as mem_window0. I think you should probably be using DT_NODELABEL instead.

Reply:
I was suspecting this would be the case, was too good to be true. Need to rework a few things then. Thanks.